### DISCUSSION TITLE
[v4]:: Default text sizes changed!

### DISCUSSION THREAD

**Discussion by Nayeem-XTREME** [tailwindlabs/tailwindcss#15397] [Help] [answered]
After migrating from tailwind v3 to v4, seems like the default font sizing (line-height) is changed. How to keep the old values, so that after migration the design look exactly same.

From bellow example:

Tailwind v3:
```
.text-xs {
    font-size: .75rem;
    line-height: 1rem;
}
```

and in Tailwind v4:
```
.text-xs {
    font-size: var(--text-xs);
    line-height: var(--tw-leading, var(--text-xs--line-height));
}
```

...............

Tailwind v4:
![image](https://github.com/user-attachments/assets/e0f105cb-d54b-4cc6-85c5-5da1c030bea9)
![image](https://github.com/user-attachments/assets/fe4daab2-c4f2-4add-ae48-fdadd635dc2b)

Tailwind v3:
![image](https://github.com/user-attachments/assets/c11a1d20-40a3-4c02-98cd-47c0ec0ee9c3)

**Comment by wongjn** [comment, marked as answer]
It should be the same:

```
line-height = 1 ÷ 0.75
            = 1.333…
            = 1.333… × 0.75rem
            ≈ 1rem
```

**Reply by liby** [reply]
After upgrading from Tailwind CSS v3 to v4, we observed subpixel rounding issues with `line-height` values in the `@tailwindcss/typography` plugin.

Both v3 and v4 source CSS contain `line-height: 1.3333333` (7 decimal places). However, the browser receives different values:

| Version | CSS received by browser | Computed result |
|---------|------------------------|-----------------|
| v3 | `1.3333333` (7 digits) | `32px` |
| v4 | `1.33333` (5 digits) | `31.9999px` |

The v4 build toolchain (likely PostCSS, Vite plugin, or CSS processing) appears to truncate numeric precision somewhere in the pipeline, while v3 preserves it.

<img width="797" height="292" alt="image" src="https://github.com/user-attachments/assets/e54152aa-9734-4371-b7a9-8a381703107a" />

This causes: `24px × 1.33333 = 31.9999px` instead of `24px × 1.3333333 ≈ 32px`

<img width="817" height="290" alt="image" src="https://github.com/user-attachments/assets/b840bfc5-9b2f-4ae2-8d48-a9da2ea11a93" />
